src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
By generalize, I meant rename `signed_integer_overflow` to something more generally describing undefined/indeterminate behavior.

Another option is to keep `signed_integer_overflow`, add more intrinsics that are similar (e.g. `divided_by_zero`), and then have a helper function that returns true if an expression is any one of these poison value intrinsics.